packages/fxa-metrics-processor/src/config.ts

```diff
@@ -0,0 +1,93 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import convict from 'convict';
+import fs from 'fs';
+import path from 'path';
+
+const conf = convict({
+  env: {
+    default: 'production',
+    doc: 'The current node.js environment',
+    env: 'NODE_ENV',
+    format: ['development', 'test', 'stage', 'production']
+  },
+  logging: {
+    app: { default: 'fxa-user-admin-server' },
+    fmt: {
+      default: 'heka',
+      env: 'LOGGING_FORMAT',
+      format: ['heka', 'pretty']
+    },
+    level: {
+      default: 'info',
+      env: 'LOG_LEVEL'
+    },
+    routes: {
+      enabled: {
+        default: true,
+        doc: 'Enable route logging. Set to false to trimming CI logs.',
+        env: 'ENABLE_ROUTE_LOGGING'
+      },
+      format: {
+        default: 'default_fxa',
+        format: ['default_fxa', 'dev_fxa', 'default', 'dev', 'short', 'tiny']
+      }
+    }
+  },
+  queues: {
+    pubsubProjectId: {
+      default: 'default-project',
+      doc: 'GCP Pub/Sub Project Id the queues are under',
+      env: 'PUBSUB_PROJECT_ID',
+      format: 'String'
+    },
+    rawEvents: {
+      default: 'fxa-metrics-raw-events',
+      doc: 'GCP Pub/Sub Queue for FxA Raw events',
+      env: 'PUBSUB_RAW_EVENT_QUEUE',
+      format: 'String'
+    },
+    deadLetterTopic: {
+      default: 'fxa-metrics-dead-letter',
+      doc: 'GCP Pub/Sub topic for FxA metric event failures',
+      env: 'PUBSUB_DEADLETTER_QUEUE',
+      format: 'String'
+    },
+    amplitudeTopic: {
+      default: 'fxa-metrics-amplitude',
+      doc: 'GCP Pub/Sub topic for transformed Amplitude events',
+      env: 'PUBSUB_AMPLITUDE_QUEUE',
+      format: 'String'
+    },
+    maxEventsPerBatch: {
+      default: 500,
+      doc: 'GCP Pub/Sub events to pull per batch',
+      env: 'PUBSUB_MAX_EVENT_BATCH',
+      format: 'Number'
+    }
+  },
+  sentryDsn: {
+    default: '',
+    doc: 'Sentry DSN for error and log reporting',
+    env: 'SENTRY_DSN',
+    format: 'String'
+  }
+});
+
+// handle configuration files.  you can specify a CSV list of configuration
+// files to process, which will be overlayed in order, in the CONFIG_FILES
+// environment variable.
+
+// Need to move two dirs up as we're in the compiled directory now
+const configDir = path.dirname(path.dirname(__dirname));
```

Comment:
Comment: I have mixed feelings about this bit at the bottom, which seems to get copied around a lot. I don't know if there's an effective way to extract it into fxa-shared or upstream it into the convict config library. Not necessarily any action to take here, but would be nice to DRY this up.

Reply:
Indeed, and at some point it could be more hassle to pull it in everywhere than copy/paste it.